### PR TITLE
Fix component dependencies

### PR DESCRIPTION
**Pull request by istoilovgithub**
Fix dependencies for parallel build
Enable parallel build on Travis

Related tasks [APPLINK-30588](https://adc.luxoft.com/jira/browse/APPLINK-30588) [APPLINK-30973](https://adc.luxoft.com/jira/browse/APPLINK-30973)

### CODE REVIEW

**Comment by LuxoftAKutsan** [comment]
@istoilovgithub i see on travis
```
CMake Error at src/3rd_party/expat-2.1.0/CMakeLists.txt:4 (project):

  The CMAKE_C_COMPILER:

    gcc-4.9

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment

  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to

  the compiler, or to the compiler name if it is in the PATH.

CMake Error at src/3rd_party/expat-2.1.0/CMakeLists.txt:4 (project):

  The CMAKE_CXX_COMPILER:

    g++-4.9
```

Is it reproduced same on local workstation on clean environment or it is local issue?

**Comment by istoilovgithub** [comment]
@LuxoftAKutsan I am unable to reproduce it on my local environment, so I  suppose that it is an issue on travis, moreover the other build check passed just fine.

File: .travis.yml [hunk 1]
```diff
@@ -33,7 +33,8 @@ before_install:
 script:
   - sudo ln -sf /usr/bin/gcov-4.9 /usr/bin/gcov
   - bash -e tools/infrastructure/check_style.sh
-  - mkdir build && cd build && cmake ../ -DBUILD_TESTS=ON -DENABLE_GCOV=ON && make install
+  - echo Number of processing units available: ${PROCESSING_UNITS_COUNT}
```
Comment (orhan-mehmedov) [inline .travis.yml]: @istoilovgithub I see strange error on travis related to this line:
{:echo Number of processing units available=: command not found
Please, check & fix.

File: src/appMain/CMakeLists.txt [hunk 2]
```diff
@@ -143,6 +143,12 @@ endif()
 add_executable(${PROJECT} ${SOURCES})
 target_link_libraries(${PROJECT} ${LIBRARIES})
 
+add_dependencies(${PROJECT} Policy)
+
+if(ENABLE_LOG)
+  add_dependencies(${PROJECT} install-3rd_party_logger)
```
Comment (orhan-mehmedov) [inline src/appMain/CMakeLists.txt]: @istoilovgithub appMain depends from Utils and must wait building of Utils to finish(and all its dependencies). Utils must take care of logger - build/install. Is it needed to add here explicit dependency of logger install?

Comment (istoilovgithub) [inline src/appMain/CMakeLists.txt]: @orhan-mehmedov this explicit dependency ensures proper logger installation (configuration, appenders, etc.), correct me if I am wrong, but I don't see an issue to have it here.

Comment (orhan-mehmedov) [inline src/appMain/CMakeLists.txt]: @istoilovgithub the idea is Utils to hide details about loggers etc. and to have once place to edit if in the future we do not depend from this logger etc.

File: .travis.yml (hunk 1 above)
Comment (istoilovgithub) [inline .travis.yml]: @orhan-mehmedov fixed with e4492fe

File: src/appMain/CMakeLists.txt (hunk 2 above)
Comment (istoilovgithub) [inline src/appMain/CMakeLists.txt]: @orhan-mehmedov agreed, let's keep it implicitly trough Utils, see e4492fe

**Pull request merged**